component/storage/edv/encryptedformatter.go

```diff
@@ -126,18 +142,46 @@ func (e *EncryptedFormatter) format(keyAndTagPrefix, key string, value []byte, t
 
 	var formattedValue []byte
 
-	if documentID != "" {
```

Comment:
Looks like you removed the `if documentID != ""` check - was that intentional? I can't remember why I put that there in the first place, so if that was an intentional change we should discuss briefly just to make sure it doesn't break something

Reply:
```
	if e.useDeterministicDocumentIDs {
		documentID, err = e.generateDeterministicDocumentID(keyAndTagPrefix, key)
		if err != nil {
			return "", nil, nil, fmt.Errorf("failed to format key into an encrypted document ID: %w", err)
		}
	} else {
		documentID, err = generateRandomDocumentID()
		if err != nil {
			return "", nil, nil, fmt.Errorf("failed to generate EDV-compatible ID: %w", err)
		}
	}
```

above will ensure `documentID` is not empty, unless one of those calls return an empty value, which is not the case.

Reply:
Makes sense!

Reply:
reverted this change to minimize impact of this pr

Reply:
Actually on closer look, `generateDeterministicDocumentID` can return a blank ID, so yeah reverting is the right thing to do